calico_node/startup/autodetection/reachaddr.go

```diff
@@ -0,0 +1,50 @@
+package autodetection
+
+import (
+	"fmt"
+	gonet "net"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// ReachDestination auto-detects the interface Network by setting up a UDP
+// connection to a "reach" destination.
+func ReachDestination(dest string, version int) (*net.IPNet, error) {
+	log.Debugf("Auto-detecting IPv%d CIDR by reaching destination %s", version, dest)
+
+	// Open a UDP connection to determine which external IP address is
+	// used to access the supplied destination.
+	protocol := fmt.Sprintf("udp%d", version)
+	address := fmt.Sprintf("[%s]:80", dest)
+	conn, err := gonet.Dial(protocol, address)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	// Get the local address as a golang IP and use that to find the matching
+	// interface CIDR.
+	addr := conn.LocalAddr()
+	if addr == nil {
+		return nil, fmt.Errorf("no address detected by connecting to %s", dest)
```

Comment:
I think more friendly error message could be something like `"no addresses found matching %s interface CIDR."`. WDYT?